userspace/libscap/engine/bpf/scap_bpf.c

```diff
@@ -86,45 +85,9 @@ struct bpf_map_data {
 	struct bpf_map_def def;
 };
 
-static const char* resolve_bpf_probe(const char *bpf_probe, char *buf)
```

Comment:
We receive the bpf path directly from the client :)

Reply:
Yeah, this probably was born as a last attempt to open something, the problem is that some consumers like Falco use this workaround as the only way to open the probe and this is really strange IMHO :thinking: On the other side with this new approach is the consumer that decides the engine and which params it must have so we cannot rely anymore on this kind of approach :/ 
And yes, definitely i forgot them, thank you!

Reply:
Well, this is the reason I added the `match` method to the vtable (each engine can look into the scap_open_args and decide if it can support them). The `bpf` engine tries to resolve the probe in its `match` method to support the current API (I'm not touching the sinsp<->scap API in my series even though I'd love to).

The method isn't used yet (in the submitted/merged PRs) but the end result would be:

```
struct scap_vtable* find_engine(struct scap_open_args *args)
{
    for(i=0; i<ARRAY_SIZE(known_engines); ++i) {
        if(known_engines[i]->match(args)) {
            return known_engines[i];
        }
    }
    return NULL; // no suitable engine found
}
```

and the rest of libscap should be completely engine-agnostic (modulo the `known_engines` array which is a weak point of the design)

Reply:
Uhm I understood, that seems a smart approach but since we are changing the sinsp interface we can solve the problem at the root. This seems the right time to perform this change since we are quite far from the libs tag and the consumers can have the right time to change the interface :thinking:

Reply:
Sure. **I**'m not touching sinsp (my PRs are big enough as is) but if you're willing to open that can of worms, go ahead :)

But my request if you do change the sinsp API would be to submit a PR to Falco at a similar time, so that consumers have at least something to base their changes on.

Reply:
ahahah yeah for sure, if we agree on that I can open a PR in Falco even before this one is merged :)